client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -78,6 +80,8 @@ type PlanRowOptions = {
 	previousProductNameShort?: string;
 };
 
+export type PlanSelectedStorage = { [ key: string ]: WPComStorageAddOnSlug | null };
```

Comment:
This type would migrate to the data store I presume, right? 🤔

Reply:
Good question -- yeah this could live in the data store if we want UI state to live there. It represents what upgrade add-on is currently selected for a given plan

Reply:
> Whether or not it should depends on what kind of vision we have for the data store and how dumb we want the actual plan-features-2023-grid component to be 🤔

ehm -- **extremely** dumb :-)

To summarise once more: we will be exporting individual parts (`PlanFeaturesGrid`, `PlanComparisonGrid`, `PlanSpotlight`, etc.). The index file (`plan-features-2023-grid`) will no longer be a wrapper component that puts everything together. We already have a wrapper in `plans-features-main`. Grid package (like the existing `plans-grid` package) will export discrete components that we import and render anywhere and however we want (in modals, sidebars, etc.)

and since these storage selections feed into more than one component (they are no longer specific to the features table), then the alternative to a `plans-wpcom-ui` data store would be to control these from a root component and send through setters and values as props (which will lead to more complex interfaces, prop drilling, potential blockers with missing an immediate root, etc. - the very thing that a data store aims to solve). 

akin to the Onboard store that serves different flows, plans-ui store will serve different plan UI components.

Reply:
that's the idea, what we aim for (and we are not far from it - these can happen prior to package extraction)

Reply:
> and since these storage selections feed into more than one component (they are no longer specific to the features table),

if I'm reading this correctly. that's the case from what I understand, right?

Reply:
> if I'm reading this correctly. that's the case from what I understand, right?

Yes! Totally right. It'll affect many components 👍